GroupByDigitNumbers.java

```diff
@@ -12,19 +13,15 @@ public class GroupByDigitNumbers {
       the output should be 1 -> ["o3", "e8"]; 2 -> ["e44", "o51"]; 3 -> ["e108"]
      */
     public Map<Integer, List<String>> groupByDigitNumbers(int[] input) {
-        Map<Integer, List<String>> map = new HashMap<>();
-
-        for (int number : input) {
-            if (number < 0) continue;
-
-            int length = (int) (Math.log10(number) + 1);  // use this instead of turning int to string then getting length
-            String prefix = (number % 2 == 0) ? "e" : "o";
-            String value = prefix + number;
-
-            map.computeIfAbsent(length, k -> new ArrayList<>()).add(value);
-        }
-
-        return map;
+        return Arrays.stream(input)
+                .filter(number -> number >= 0).mapToObj(number -> {
+                    String prefix = (number % 2 == 0) ? "e" : "o";
+                    return prefix + number;
+                })
+                .collect(Collectors.groupingBy(
+                        str -> str.length()-1,
+                        Collectors.toList()
```

Comment:
no need in the second argument, just Collectors.groupingBy(str -> str.length()-1) should work